packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart

```diff
@@ -137,13 +140,15 @@ class MethodChannelShare extends SharePlatform {
       final tempSubfolderPath = "$tempRoot/${const Uuid().v4()}";
       await Directory(tempSubfolderPath).create(recursive: true);
 
+      //Per Issue [#3032](https://github.com/fluttercommunity/plus_plugins/issues/3032): use overriden name when avaiable.
       //Per Issue [#1548](https://github.com/fluttercommunity/plus_plugins/issues/1548): attempt to use XFile.name when available
-      final filename = file.name.isNotEmpty // If filename exists
-              ||
-              lookupMimeType(file.name) !=
-                  null //If the filename has a valid extension
-          ? file.name
-          : "${const Uuid().v1().substring(10)}.$extension";
+      final filename = nameOverride ??
+          (file.name.isNotEmpty // If filename exists
+                  ||
+                  lookupMimeType(file.name) !=
+                      null //If the filename has a valid extension
+              ? file.name
+              : "${const Uuid().v1().substring(10)}.$extension");
```

Comment:
I know this was not your code, but since we are at it, could this be simplified a bit?

Reply:
I have factored out the `file.name.isNotEmpty    ||   lookupMimeType(file.name) !=  null` boolean in a separate variable to simplify the expression.

However I'm not sure if this check makes too much sense in the first place.
In a scenario where the filename is not empty, the left half is true and the right doesn't get evaluated.
In a scenario there the filename *is* empty I'm pretty sure `lookupMimeType('')` will also return null.

Reply:
Had a second look and I think it is fine now, thanks!